discord_connector.py

```diff
@@ -0,0 +1,436 @@
+
+    def fetch_message(self, channel_id, message_id, action_result) -> discord.Message or None:
+        status, channel = self.run_in_loop(self._guild.fetch_channel(channel_id), action_result,
+                                           error_message=DISCORD_ERROR_FETCHING_CHANNEL)
+        if not status:
+            return status, None
+        status, message = self.run_in_loop(channel.fetch_message(message_id), action_result,
+                                           error_message=DISCORD_ERROR_FETCHING_MESSAGE)
+
+        return status, message
+
+    def create_artifacts(self, message):
+        container_id = self.get_container_id()
+        attachments = []
+        embeds = []
+
+        if message.embeds:
+            self.save_progress("working on embeds")
+            for embed in message.embeds:
+                embeds.append(self.create_embed_artifact(embed, container_id))
+
+        if message.attachments:
+            self.save_progress("working on attachments")
+            for attachment in message.attachments:
+                attachments.append(self.create_attachment_artifact(attachment, container_id))
+
+        return attachments, embeds
+
+    def create_embed_artifact(self, embed, container_id):
+        artifact = Artifact(
+            container_id=container_id,
+            name=f"embed: {embed.title}",
+            cef={"URL": embed.url, "Description": embed.description}
+        )
+        return self.save_artifact_to_soar(dataclasses.asdict(artifact))
+
+    def create_attachment_artifact(self, attachment, container_id):
+        artifact = Artifact(
+            container_id=container_id,
+            name=f"attachment: {attachment.filename}",
+            cef={"URL": attachment.url, "Description": attachment.description, "Type": attachment.content_type}
+        )
+        return self.save_artifact_to_soar(dataclasses.asdict(artifact))
+
+    def save_artifact_to_soar(self, artifact):
+        status, creation_message, artifact_id = self.save_artifact(artifact)
+        self.save_progress("creating artifact: status: {}, creation message: {}, artifact id {}"
+                           .format(status, creation_message, artifact_id))
+        return artifact_id
+
+    def parse_message(self, message, attachments, embeds):
+        return {
+            "message origin": {
+                "channel id": message.channel.id,
+                "channel name": message.channel.name,
+            },
+            "message data": {
+                "created at": str(message.created_at),
+                "edited at": str(message.edited_at) if message.edited_at is not None else "message was not edited",
+            },
+            "author data": {
+                "author id": message.author.id,
+                "author name": message.author.name,
+            },
+            "jump url": message.jump_url,
+            "flags": [flag[:1] for flag in list(filter((lambda flag: flag[1]), message.flags))] or "no flags",
```

Comment:
why :1 ? Isn't it just 0?

Reply:
:1 is there becouse the result of list(filter(lamba is list of tuples from with we need only the first value

Reply:
I just figured out what you ment by :1, I am going to change that